server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java

```diff
@@ -555,7 +557,7 @@ public interface PersistedState extends Closeable {
          * with the last committed configuration now corresponding to the last accepted configuration, and the cluster uuid, if set,
          * marked as committed.
          */
-        default void markLastAcceptedStateAsCommitted() {
+        default void markLastAcceptedStateAsCommitted(DiscoveryNode sourceNode, long term, long version) {
```

Comment:
AFAICT these additional arguments are only used to assert that the state we're committing is the one we expect it to be. If we need such an assertion, I think it'd be better to assert that just before calling this method (i.e. in  org.elasticsearch.cluster.coordination.CoordinationState#handleCommit). That way we can drop these args.